onnxruntime/core/providers/qnn/qnn_execution_provider.cc

```diff
@@ -121,14 +120,6 @@ QNNExecutionProvider::QNNExecutionProvider(const ProviderOptions& provider_optio
     ParseHtpPerformanceMode(htp_performance_mode_pos->second);
   }
 
-  AllocatorCreationInfo device_info(
-      [](int) {
-        return std::make_unique<CPUAllocator>(OrtMemoryInfo(QNN, OrtAllocatorType::OrtDeviceAllocator));
-      });
-
-  cpu_allocator_ = CreateAllocator(device_info);
-  InsertAllocator(cpu_allocator_);
```

Comment:
This is common for almost all EPs. Don't you need to update it for other EPs?

Reply:
There is another PR https://github.com/microsoft/onnxruntime/pull/15833 targetting all the EPs. QNN is kind of special, we can use local CPU Allocator instead of the class member cpu_allocator_, which is not true for most of the EPs.

Reply:
For more information, PR https://github.com/microsoft/onnxruntime/pull/15833 aims to detach allocator from EP and move them to session level. Currently some EPs don't have their own allocator instance, some have, but only use it inside the GetAllocator() function, which will be replaced in a session level function in the above PR. 

QNN, however, has its member allocator and use it in compile() function which quite different from most of the EP, that is why I start a seperate PR to do the refactor, as PR 15833 is already very huge, I'd like to separate everything possible out of the PR 15833 to make it easier for me

Reply:
sounds good.

---
In reply to: [1206102944](https://github.com/microsoft/onnxruntime/pull/16084#discussion_r1206102944) [](ancestors = 1206102944)